internal/processor/filterset/factory/config.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package factory
+
+const (
+	// REGEXP is the FilterType for filtering by regexp string matches.
+	REGEXP MatchType = "regexp"
+	// STRICT is the FilterType for filtering by exact string matches.
+	STRICT MatchType = "strict"
+)
+
+// MatchType describes the type of pattern matching a FilterSet uses to filter strings.
+type MatchType string
+
+// MatchConfig configures the matching behavior of a FilterSet.
+type MatchConfig struct {
+	MatchType MatchType     `mapstructure:"match_type"`
+	Regexp    *RegexpConfig `mapstructure:"regexp"`
+	Strict    *StrictConfig `mapstructure:"strict"`
+}
+
+// StrictConfig is used to configure options for NewStrictFilterSet.
+type StrictConfig struct{}
```

Comment:
This empty struct does not seem to serve any purpose as there is nothing that can be specified under the "strict" subkey. Why do we need it?

Reply:
I was going for consistency with the regexpconfig so that the user would still have a `strict:` section in the config.yaml, but it's just probably best to just remove it. Removed.